apps/datapuller/src/pullers/classes.ts

```diff
@@ -10,6 +10,73 @@ import {
 
 const TERMS_PER_API_BATCH = 4;
 
+const checkTermHasCatalogData = async (
+  year: number,
+  semester: string,
+  academicCareerCode: string
+): Promise<boolean> => {
+  const termExists = await TermModel.exists({
+    name: `${year} ${semester}`,
+    academicCareerCode,
+  });
+
+  if (!termExists) return false;
+
+  const classExists = await ClassModel.exists({
+    year,
+    semester,
+  });
```

Comment:
this should be the same as the getCatalog query:
```js
{
    year,
    semester,
    anyPrintInScheduleOfClasses: true,
}
```

We can also do `countDocuments()` and make sure that's greater than some low threshold, like 100 or something.